aws_topology/stackstate_checks/aws_topology/resources/autoscaling.py

```diff
@@ -83,16 +83,19 @@ def process_autoscaling_group(self, data):
             self.emit_relation(auto_scaling_group.AutoScalingGroupARN, instance.InstanceId, 'uses service', {})
 
         for load_balancer_name in auto_scaling_group.LoadBalancerNames:
+            elb_arn = self.agent.create_arn(
+                "AWS::ElasticLoadBalancing::LoadBalancer", self.location_info, resource_id=load_balancer_name
```

Comment:
Is there not a way to re-use the loadbalancer arn function that is already defined?

Reply:
That's what this does, but it goes through the collector first. It does actually call the create_arn at the top of the file, but it's done this way because many times we need to do cross-API calls to create an ARN (for example when creating a relation) so we use the same method everywhere.

So the entire thing is abstracted into the collector. Normally we get the ARN back in the API call, but loadbalancing is an old API and is an exception

Reply:
I am planning to change how it all works in a later stage. The difference between using the agent and using the method directly is that we change change that around more easily. The agent will always know how to resolve, the implementaion might change in future.